library/x509_crt.c

```diff
@@ -820,7 +820,15 @@ static int x509_get_crt_ext( unsigned char **p,
             break;
 
         default:
-            return( MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE );
+            /*
+             * In case this is a non critical extension, which the oid layer
+             * support, but there isn't an x509 parser for it,
+             * skip the extension.
+             */
+            if( is_critical )
+                return( MBEDTLS_ERR_X509_FEATURE_UNAVAILABLE );
+            else
+                *p = end_ext_octet;
```

Comment:
Should this be wrapped in `#if !defined(MBEDTLS_X509_ALLOW_UNSUPPORTED_CRITICAL_EXTENSION)`?

Reply:
Yes, I  believe you are right.